curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentNode.java

```diff
@@ -462,7 +484,12 @@ else if ( existingPath != null && mode.isSequential() )
             if ( localCreateMethod == null )
             {
                 CreateBuilderMain createBuilder = mode.isTTL() ? client.create().withTtl(ttl) : client.create();
-                CreateModable<ACLBackgroundPathAndBytesable<String>> tempCreateMethod = useProtection ? createBuilder.creatingParentContainersIfNeeded().withProtection() : createBuilder.creatingParentContainersIfNeeded();
+                CreateModable<ACLBackgroundPathAndBytesable<String>> tempCreateMethod;
+                if (useParentCreation) {
+                    tempCreateMethod = useProtection ? createBuilder.creatingParentContainersIfNeeded().withProtection() : createBuilder.creatingParentContainersIfNeeded();
```

Comment:
I found that this part of code is a bit awkward since our fluent api can flow only one directory and not quite fluent when one needs do configuration one by one. Is there a best practice for doing sth like

```
if (useParentCreate) {
  tempCreateMethod = tempCreateMethod.creatingParentContainersIfNeeded();
}
if (useProtection) {
  tempCreateMethod = tempCreateMethod.withProtection();
}
```

Reply:
I can't think of any workaround to do that in a better way 😕

Reply:
@Randgalt any ideas for this case?

Reply:
I've done things like that before. It seems OK